apps/test-viewer/src/UseAccessToken.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useEffect, useState } from "react";
+import { AuthorizationState, useAuthorizationContext } from "./components/Authorization";
+
+export function useAccessToken() {
+  const { state, client } = useAuthorizationContext();
+  const [accessToken, setAccessToken] = useState<string | undefined>(undefined);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchAccessToken = async () => {
+      if (state === AuthorizationState.Pending) {
+        setAccessToken(undefined);
+        setLoading(true);
+        return;
+      }
+
+      setLoading(true);
```

Comment:
I think that we can `setLoading` at the start, since it is set to true in `if` block as well as after it
```suggestion
    const fetchAccessToken = async () => {
      setLoading(true);
      if (state === AuthorizationState.Pending) {
        setAccessToken(undefined);
        return;
      }

```

Reply:
After refactor loading was removed as it was not used.